lib/taskana-core-test/src/test/java/acceptance/task/update/UpdateTaskAccTest.java

```diff
@@ -0,0 +1,561 @@
+          throws Exception {
+    List<Pair<String, ObjectReference>> list =
+        List.of(
+            Pair.of("When Primary Object Reference is null", null),
+            Pair.of(
+                "When Value of Primary Object Reference is null",
+                createObjectReference("COMPANY_A", "SYSTEM_A", "INSTANCE_A", "VNR", null)),
+            Pair.of(
+                "When Type of Primary Object Reference is null",
+                createObjectReference("COMPANY_A", "SYSTEM_A", "INSTANCE_A", null, "1234567")),
+            Pair.of(
+                "When Company of Primary Object Reference is null",
+                createObjectReference(null, "SYSTEM_A", "INSTANCE_A", "VNR", "1234567")));
+
+    ThrowingConsumer<Pair<String, ObjectReference>> test =
+        pair -> {
+          Task task =
+              TaskBuilder.newTask()
+                  .classificationSummary(defaultClassificationSummary)
+                  .workbasketSummary(defaultWorkbasketSummary)
+                  .primaryObjRef(defaultObjectReference)
+                  .buildAndStore(taskService);
+          task.setPrimaryObjRef(pair.getRight());
+          assertThatThrownBy(() -> taskService.updateTask(task))
+              .isInstanceOf(InvalidArgumentException.class);
+        };
+    return DynamicTest.stream(list.iterator(), Pair::getLeft, test);
+  }
+
+  @WithAccessId(user = "user-taskrouter")
+  @Test
+  void should_ThrowException_When_UserIsNotAuthorized() throws Exception {
+    Task task =
+        TaskBuilder.newTask()
+            .classificationSummary(defaultClassificationSummary)
+            .workbasketSummary(defaultWorkbasketSummary)
+            .primaryObjRef(defaultObjectReference)
+            .buildAndStore(taskService, "admin");
+    task.setPrimaryObjRef(
+        taskService.newObjectReference("company", "system", "instance", "value", "type"));
+    ClassificationSummary classificationSummary =
+        defaultTestClassification().buildAndStoreAsSummary(classificationService, "admin");
+    task.setClassificationKey(classificationSummary.getKey());
+
+    ThrowingCallable call = () -> taskService.updateTask(task);
+
+    NotAuthorizedOnWorkbasketException e =
+        catchThrowableOfType(call, NotAuthorizedOnWorkbasketException.class);
+    assertThat(e.getCurrentUserId()).isEqualTo("user-taskrouter");
+    assertThat(e.getWorkbasketId()).isEqualTo(defaultWorkbasketSummary.getId());
+    assertThat(e.getRequiredPermissions()).containsExactly(WorkbasketPermission.READ);
+  }
+
+  @WithAccessId(user = "user-1-2")
+  @Test
+  void should_ThrowException_When_TaskHasAlreadyBeenUpdated() throws Exception {
+    Task task =
+        TaskBuilder.newTask()
+            .classificationSummary(defaultClassificationSummary)
+            .workbasketSummary(defaultWorkbasketSummary)
+            .primaryObjRef(defaultObjectReference)
+            .buildAndStore(taskService, "admin");
+    final Task task2 = taskService.getTask(task.getId());
```

Comment:
You don't need final here. Please use copy() instead of getTask

Reply:
Final has to be used, since distance between declarations and first usage is more than 3 lines (checkstyle violation)
Using .copy() with final declaration causes all task attributes to be set to null, so getTask has to be used